### DISCUSSION TITLE
Authentication using encrypted private key

### DISCUSSION THREAD

**Discussion by yohei1126** [Snowflake-Labs/terraform-provider-snowflake#440] [Q&A] [answered]
Does this support authentication using encrypted private key?

I have a provider configuration like this.

```
provider "snowflake" 
  username = "terraform"
  account  = "xxx"
  region   = "ap-southeast-1"
  private_key_path = "rsa_key.p8"
  role             = "xxx"
}
```

I got this error when I tried to run `terraform plan`.

```
$ AWS_PROFILE=xxx SNOWSQL_PRIVATE_KEY_PASSPHRASE="yyy"  terraform plan --out out.tfplan

Error: could not build dsn for snowflake connection: Private Key could not be parsed: Could not parse private key: ssh: unsupported key type "ENCRYPTED PRIVATE KEY"
```

**Comment by yohei1126** [comment, marked as answer]
@ryanking @alldoami Any thoughts?

**Reply by ryanking** [reply]
It is not currently supported. I would be happy to review a PR for this (or make @alldoami review it :)).

Note - I think it might be hard to get the interface right for this, in a way that is both secure and usable.

**Reply by yohei1126** [reply]
Strange. Snowflake CLI and SDK officially support encrypted private key.
